app/tasks/raster_tile_set_assets/raster_tile_set_assets.py

```diff
@@ -48,10 +63,116 @@ async def raster_tile_set_asset(
 
     callback: Callback = callback_constructor(asset_id)
 
-    create_raster_tile_set_job = await create_pixetl_job(
+    create_raster_tile_set_job: Job = await create_pixetl_job(
         dataset, version, creation_options, "create_raster_tile_set", callback
     )
 
     log: ChangeLog = await execute([create_raster_tile_set_job])
 
     return log
+
+
+async def get_extent(asset_id: UUID) -> Extent:
+    asset_row: ORMAsset = await get_asset(asset_id)
+    asset_uri: str = get_asset_uri(
+        asset_row.dataset,
+        asset_row.version,
+        asset_row.asset_type,
+        asset_row.creation_options,
+        srid=infer_srid_from_grid(asset_row.creation_options.get("grid")),
+    )
+    bucket, key = split_s3_path(tile_uri_to_extent_geojson(asset_uri))
+
+    s3_client = get_s3_client()
+    resp = s3_client.get_object(Bucket=bucket, Key=key)
+    extent_geojson: Dict[str, Any] = json.loads(resp["Body"].read().decode("utf-8"))
+
+    return Extent(**extent_geojson)
+
+
+def _create_or_augment_histogram(
+    source_histo_i: Dict[str, Any], target_histo_i: Optional[Histogram]
+):
+    if target_histo_i is None:
+        target_histo_i = Histogram(
+            **{
+                "min": source_histo_i["min"],
+                "max": source_histo_i["max"],
+                "bin_count": source_histo_i["count"],
+                "value_count": source_histo_i["buckets"],
+            }
+        )
+    else:
+        target_histo_i.min = min(target_histo_i.min, source_histo_i["min"])
+        target_histo_i.max = max(target_histo_i.max, source_histo_i["max"])
+        for bucket_num, bucket_val in enumerate(source_histo_i["buckets"]):
+            target_histo_i.value_count[bucket_num] += bucket_val
```

Comment:
unfortunately this won't work all the time.

GDAL always returns a histogram with 256 buckets. Bucket size is defined by `min`, `max` and data type.

Ie if `min=0` and `max=1` and `datatype=int`, bucket size will be 1. Hence the first bucket will list the count for all `0`, the second bucket will list the counts for all `1` and the remaining 254 buckets will be empty.

if we have a `min=0` and `max=1000` and we have any values in between, it will be more tricky. In that case, bucket size will be most likely `4` and values will be grouped together in their corresponding bucket. Ie values `0-3` will be counted in the first bucket, values `4-7` in the second and so on.

If we now try to merge to histograms, we will have to make sure that the bucket size and the min/ max are the same, otherwise buckets won't line up. Ie, one raster might have a bucket size of 3 and a range of 200 - 900 and the other one a size of 4 and a range of 0 - 1000.

A possible solution on how to address this issue is described here
https://stackoverflow.com/a/47183002/1410317